app/projects/forms.py

```diff
@@ -734,8 +734,11 @@ def __init__(self, *args, **kwargs):
         if "input_timeseries" in self.fields:
             self.fields["input_timeseries"] = TimeseriesField(
                 qs_ts=Timeseries.objects.filter(
-                    Q(ts_type=self.asset_type.mvs_type)
-                    & (Q(open_source=True) | Q(user=self.user))
+                    (
+                        Q(ts_type=self.asset_type.mvs_type)
+                        & Q(component=self.asset_type.asset_type)
+                    )
+                    & (Q(open_source=True) | Q(project__id=proj_id))

```

Comment:
```suggestion
                    & (Q(open_source=True) | Q(scenario__project__id=proj_id))

```
We could also get the project ID through the scenario, that way we wouldn't need to add the project field to timeseries (it seems redundant to have both `project` and `scenario` since they are linked anyways).

Reply:
Yes! That makes more sense!